src/components/canvas-tools/geometry-tool.tsx

```diff
@@ -583,14 +583,14 @@ class GeometryToolComponentImpl extends BaseComponent<IProps, IState> {
 
         // click on selected element - deselect if appropriate modifier key is down
         if (geometryContent.isSelected(id)) {
-          if (evt.ctrlKey || evt.metaKey) {
+          if (hasSelectionModifier(evt)) {
```

Comment:
Did you intend to change the behavior here or should this be `hasDiscontiguousModifier`? I'm not completely clear on the names of those functions.

Reply:
It's an intentional change. When I wrote that code originally I was thinking there would be a difference between contiguous (shift) and discontiguous (ctrl/cmd) selection, but I eventually decided that the distinction didn't make sense in the geometry tool.

Reply:
OK! Is it worth getting rid of those utility functions then?

Reply:
From an API perspective I think it makes sense to have the complete set, even if they're not currently used.